### PR TITLE
dev/core#2790 Move pdf postProcess function to the trait

### PR DESCRIPTION
**Pull request by eileenmcnaughton**
Overview
----------------------------------------
dev/core#2790 Move pdf postProcess function to the trait

Before
----------------------------------------
Functionality on the pdfCommon status class

After
----------------------------------------
Functionality in the trait

Technical Details
----------------------------------------
This just leaves the `processMessageTemplate` function to bring over

Comments
----------------------------------------
@demeritcowboy once it's all on the trait I'll figure out that liveMode regression you logged

I also put up https://github.com/civicrm/civicrm-core/pull/21478 that takes this & goes that last little stretch....

### CODE REVIEW

File: CRM/Case/Form/Task/PDF.php [hunk 1]
```diff
@@ -47,7 +47,7 @@ public function preProcess() {
    * Process the form after the input has been submitted and validated.
    */
   public function postProcess() {
-    CRM_Contact_Form_Task_PDFLetterCommon::postProcess($this);
+    $this->postProcessCommon();
```
Comment (demeritcowboy) [inline CRM/Case/Form/Task/PDF.php:50]: `Error: Call to undefined method CRM_Case_Form_Task_PDF::postProcessCommon() in CRM_Case_Form_Task_PDF->postProcess() (line 50 of ...\CRM\Case\Form\Task\PDF.php`

Comment (eileenmcnaughton) [reply CRM/Case/Form/Task/PDF.php]: @demeritcowboy arg - I added that & then meant to remove as I changed my mind about the fn name - fixed now

**Comment by demeritcowboy** [comment]
Activity print/merge crashes. It seems to have partially broken before this (it downloads but has token codes in the output), and here it crashes.

Otherwise the PR seems ok.

**Comment by eileenmcnaughton** [comment]
@demeritcowboy which tokens - I just found one rendered `{activity.activity_date_time} {activity.subject}` and the first token rendered but not the second - no crash

**Comment by eileenmcnaughton** [comment]
@demeritcowboy I just tested the following (from `CRM_Activity_Form_Task_PDFLetterCommonTest::testCreateDocumentBasicTokens`  on both master & agains this PR & it worked.

I think there are issues where a bunch of tokens that don't work are advertised but I think it's not changed by this PR (digging)

```
Subject: {activity.subject}
Date: {activity.activity_date_time}
Duration: {activity.duration}
Location: {activity.location}
Details: {activity.details}
Status ID: {activity.status_id}
Status: {activity.status}
Activity Type ID: {activity.activity_type_id}
Activity Type: {activity.activity_type}
Activity ID: {activity.activity_id}
```

**Comment by demeritcowboy** [comment]
Hmm it 500's on the test site here but yeah it seems ok elsewhere (except as you note some activity tokens not rendering). I don't see anything relevant merged since this site was built. Maybe a specific configuration? I can't see the web server log here so can't see what the actual error is.

**Comment by eileenmcnaughton** [comment]
I'm trying separately to get this working as output - but not as part of this PR



```

Subject: {activity.subject}
Date: {activity.activity_date_time}
Duration: {activity.duration}
Location: {activity.location}
Details: {activity.details}
Status ID: {activity.status_id}
(legacy) Status: {activity.status}
Status: {activity.status_id:label}
Activity Type ID: {activity.activity_type_id}
(legacy) Activity Type: {activity.activity_type}
Activity Type: {activity.activity_type_id:label}
Activity ID: {activity.activity_id}
(legacy) Activity ID: {activity.id}

```

**Comment by demeritcowboy** [comment]
Ok I can't see what the problem is so let's go with this and see if it comes up again.

**Comment by eileenmcnaughton** [comment]
@demeritcowboy I was starting to think maybe it was a bug in core that we have fixed since the PR site was generated

**Pull request merged**